internal/database/lock.go

```diff
@@ -93,40 +100,46 @@ func makeMultiUnlockFn(ctx context.Context, db *DB, lockIDs []string, expires ti
 // UnlockFn that can be used to unlock the lock. ErrAlreadyLocked will be
 // returned if there is already a lock in use.
 func (db *DB) Lock(ctx context.Context, lockID string, ttl time.Duration) (UnlockFn, error) {
+	logger := logging.FromContext(ctx).Named("database.Lock").
+		With("lock_id", lockID)
+
 	var expires time.Time
-	err := db.SerializableTx(ctx, func(tx pgx.Tx) error {
-		row := tx.QueryRow(ctx, `
-			SELECT AcquireLock($1, $2)
-		`, lockID, int(ttl.Seconds()))
+	if err := db.InTx(ctx, pgx.ReadCommitted, func(tx pgx.Tx) error {
+		row := tx.QueryRow(ctx, `SELECT AcquireLock($1, $2)`, lockID, int32(ttl.Seconds()))
+
 		if err := row.Scan(&expires); err != nil {
-			return err
+			return fmt.Errorf("failed to scan lock.expires: %w", err)
 		}
+
 		if expires.Before(thePast) {
 			return ErrAlreadyLocked
 		}
 		return nil
-	})
-	if err != nil {
+	}); err != nil {
 		return nil, err
 	}
-	logging.FromContext(ctx).Debugf("Acquired lock %q", lockID)
+
+	logger.Debugw("acquired lock")
 	return makeUnlockFn(ctx, db, lockID, expires), nil
 }
 
 func makeUnlockFn(ctx context.Context, db *DB, lockID string, expires time.Time) UnlockFn {
+	logger := logging.FromContext(ctx).Named("database.Unlock").
+		With("lock_id", lockID)
+
 	return func() error {
-		return db.SerializableTx(ctx, func(tx pgx.Tx) error {
-			row := tx.QueryRow(ctx, `
-				SELECT ReleaseLock($1, $2)
-			`, lockID, expires)
+		return db.InTx(ctx, pgx.ReadCommitted, func(tx pgx.Tx) error {
+			row := tx.QueryRow(ctx, `SELECT ReleaseLock($1, $2)`, lockID, expires)
 			var released bool
 			if err := row.Scan(&released); err != nil {
-				return err
+				return fmt.Errorf("failed to scan lock.released: %w", err)
 			}
-			if !released {
-				return fmt.Errorf("cannot delete lock %q that no longer belongs to you; it likely expired and was taken by another process", lockID)
```

Comment:
⚠️ Behavior change. It feels wrong (and makes testing _very_ difficult) to return an error if the lock was failed to release. With the new function, releasing the lock will _always_ succeed if the procedure finishes.